lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/Transactions.hs

```diff
@@ -244,101 +294,114 @@ spec = do
+        , expectFieldEqual walletName "1st Wallet"
+        , expectFieldEqual balanceAvailable 0
+        , expectFieldEqual balanceTotal 0
+        ]
+    let wDest = getFromResponse Prelude.id r0
+    let (Right seed) = fromMnemonic @'[15] @"seed" mnemonics15
+    let pwd = Passphrase $ BA.convert $ T.encodeUtf8 password
+    let rootXPrv = generateKeyFromSeed (seed, mempty) pwd
+    let st = mkSeqState @(Jormungandr 'Testnet)
+             (rootXPrv, pwd) defaultAddressPoolGap
+    let (addrD, st') = genChange pwd st
+    addrs <- listAddresses ctx wDest
+    let destination = (addrs !! 1) ^. #id
+    let amt = 100 :: Natural
+    let payload = Json [json|{
                 "payments": [{
-                    "address": #{destination},
-                    "amount": {
-                        "quantity": #{amt},
-                        "unit": "lovelace"
-                    }
-                }],
-                "passphrase": "Secure Passphrase"
-            }|]
-        let (feeMin, _feeMax) = ctx ^. feeEstimator $ TxDescription
-                { nInputs = 1
-                , nOutputs = 1
+                "address": #{destination},
+                "amount": {
+                    "quantity": #{amt},
+                    "unit": "lovelace"
                 }
-        r1 <- request @(ApiTransaction t) ctx (postTxEp wSrc) Default payload
-        verify r1
-            [ expectSuccess
-            , expectResponseCode HTTP.status202
-            ]
-        r2 <- request @ApiWallet ctx (getWalletEp wDest) Default Empty
-        verify r2
-            [ expectSuccess
-            , expectEventually ctx balanceAvailable amt
-            ]
+            }],
+            "passphrase": "Secure Passphrase"
+        }|]
+    let (feeMin, _feeMax) = ctx ^. feeEstimator $ TxDescription
+            { nInputs = 1
+            , nOutputs = 1
+            }
+    r1 <- request @(ApiTransaction t) ctx (postTxEp wSrc) Default payload
+    verify r1
+        [ expectSuccess
+        , expectResponseCode HTTP.status202
+        ]
+    r2 <- request @ApiWallet ctx (getWalletEp wDest) Default Empty
+    verify r2
+        [ expectSuccess
+        , expectEventually ctx balanceAvailable amt
+        ]
+
+    --at this point wDest wallet should have 100 in one of its address
+    --now we try to prepare (tx, [txWintess]), encode it and send 1 to
+    --empty wallet but using external-transactions endpoint
+    txsSrc <- listAllTransactions ctx wSrc
+    let txSrc:_ = filter (\tx ->
+                              any (\(AddressAmount _ amnt) ->
+                                       amnt == Quantity amt)
+                              $ NE.toList $ tx ^. #outputs
+                         ) txsSrc
+    let txHash = getApiT $ txSrc ^. #id
+    let out1@(AddressAmount _ amnt1):out2:_ =
+            NE.toList $ txSrc ^. #outputs
+    let (ix, out) = if amnt1 == Quantity amt then
+            (0::Word32, out1)
+            else
+            (1::Word32, out2)
```

Comment:
This formatting is weird, because the `else` keyword is on its own line, and is further to the left than the `if`.

I would recommend:
```hs
    let (ix, out) =
        if amnt1 == Quantity amt
        then (0::Word32, out1)
        else (1::Word32, out2)
```

Reply:
corrected